libslide/src/bignum/sub.rs

```diff
@@ -0,0 +1,145 @@
+#![allow(clippy::suspicious_arithmetic_impl)]
+use crate::bignum::utils::vec_i8_into_u8;
+use crate::bignum::utils::vec_u8_into_i8;
+use crate::bignum::Bignum;
+use std::mem;
+use std::ops;
+
+impl ops::Sub for Bignum {
+    type Output = Bignum;
+
+    fn sub(self, other: Bignum) -> Bignum {
+        // make the decimal vectors the same size
+        // fill lhs and rhs vectors as well
+        let self_size: usize = self.dec.len();
+        let other_size: usize = other.dec.len();
+        let (res_int, lhs_int, res_dec, lhs_dec) = if self_size > other_size {
+            let mut lhs_dec = vec![0; self_size];
+            lhs_dec[..other_size].clone_from_slice(&other.dec[..other_size]);
+            (self.int, other.int, self.dec, lhs_dec)
+        } else {
+            let mut lhs_dec = vec![0; other_size];
+            lhs_dec[..self_size].clone_from_slice(&self.dec[..self_size]);
+            (other.int, self.int, other.dec, lhs_dec)
+        };
+
+        // now we need to make the bigger number the res dec
+
+        let mut lhs = Bignum {
+            is_neg: false,
+            dec: lhs_dec,
+            int: lhs_int,
+        };
+        let mut rhs = Bignum {
+            is_neg: false,
+            dec: res_dec,
+            int: res_int,
+        };
```

Comment:
let's not construct a new bignum just to do the swap

Reply:
It’s not to do the swap, it’s to compare. What’s a better way of doing this then.

Reply:
nevermind, sounds good 👍

Reply:
Actually, hold up. Both self and rhs are moved in at the beginning of this function, so why not just compare them then before you construct the vectors?

Reply:
That’s how I originally wrote sub, but Then you run into an issue with adding the zeros to make the decimals the same length.

Reply:
What's wrong with that?